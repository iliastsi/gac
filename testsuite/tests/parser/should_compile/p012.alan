MulDimArray() : proc
  i : int [4][5];

  foo2(f : reference int[][5]) : proc
  { f[1][3] = 42; }
  foo3(f : reference int[4][5]) : proc
  {}
  foo5(f : reference int[5]) : proc
  {}
{
  i[1][2] = 42;

  foo2(i);
  foo3(i);
  foo5(i[3]);
}
