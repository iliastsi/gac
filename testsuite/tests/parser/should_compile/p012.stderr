==================== Parser ====================
MulDimArray() : proc
  i : int[4][5];
  foo2(f : reference int[][5]) : proc
  { -- foo2
    f[1][3] = 42;
  } -- foo2
  foo3(f : reference int[4][5]) : proc
  { -- foo3
  } -- foo3
  foo5(f : reference int[5]) : proc
  { -- foo5
  } -- foo5
{ -- MulDimArray
  i[1][2] = 42;
  foo2(i);
  foo3(i);
  foo5(i[3]);
} -- MulDimArray


